src/pallene/c_compiler.lua

```diff
@@ -61,6 +61,8 @@ function c_compiler.compile_o_to_so(in_filename, out_filename)
         CFLAGS_SHARED,
         "-o", util.shell_quote(out_filename),
         util.shell_quote(in_filename),
+        "-lptracer",
+        "-Wl,-rpath=/usr/local/lib",
```

Comment:
Do we really need rpath? I'm worried about hardcoding user/local/lib

Reply:
Yes.

Reply:
Why yes? In my system, "/usr/local/lib" is part of the default search path. And presumably the user should have installed ptracer somewhere in their default search path. Also, that directory is configurable in the ptracer makefile, so we should not be hardcoding it here.

Reply:
Removed the hardcoded part. I don't know whats up with the distro you are using, from my experience, you need some -rpath to defined. And here, my experience includes, Manjaro, Debian 10 and later and ArchLinux (which I use BTW). It does not work for every distro, I can assure you that much.

Reply:
Rpath should only be necessary if you're installing the shared library in a non-standard location, and don't want to use LD_LIBRARY_PATH.

You can check the search path in your distro by running  `ld --verbose | grep SEARCH_DIR | sed -e 's/; /\n/g'`. On mine, it includes /usr/local/lib

Reply:
It includes `/usr/local/lib` in my case as well, but still fails to link `libptracer.so` :)

Reply:
Did you try removing the entire "-Wl,-rpath=/usr/local/lib" line?

Reply:
Yes

Reply:
For your convenience, this get this error while running the test suite after removing the line.

```
Failure → ./spec/execution_tests.lua @ 103
#c_backend / Exported functions / exports public functions
./spec/execution_tests.lua:89: command failed: lua '__test_c__script.lua' > '__test_c__output.txt'

stack traceback:
	./spec/execution_tests.lua:89: in upvalue 'run_test'
	./spec/execution_tests.lua:104: in function <./spec/execution_tests.lua:103>

lua: error loading module '__test_c__' from file './__test_c__.so':
	libptracer.so: cannot open shared object file: No such file or directory
stack traceback:
	[C]: in ?
	[C]: in function 'require'
	__test_c__script.lua:1: in main chunk
	[C]: in ?
●✱
1 success / 0 failures / 1 error / 0 pending : 0.211186 seconds
```